search/search_indexes.py

```diff
@@ -33,14 +43,46 @@ def index_queryset(self, using=None):
 
     def prepare_text(self, obj):  # pylint: disable=no-self-use
         """Indexing of the primary content of a LearningResource."""
+        try:
+            # Strip XML tags from content before indexing.
+            tree = etree.fromstring(obj.content_xml)
+            content = etree.tostring(tree, encoding="utf-8", method="text")
+        except etree.XMLSyntaxError:
+            # For blank/invalid XML.
+            content = obj.content_xml
         return "{0} {1} {2}".format(
-            obj.title, obj.description, obj.content_xml,
+            obj.title, obj.description, content,
         )
 
     def prepare_run(self, obj):  # pylint: disable=no-self-use
         """Define what goes into the "run" index."""
         return obj.course.run
 
-    def prepare_course(self, obj):  # pylint: disable=no-self-use
+    @staticmethod
+    def prepare_course(obj):
         """Define what goes into the "course" index."""
         return obj.course.course_number
+
+    def prepare(self, obj):
+        """
+        Get dynamic vocabularies.
+
+        The prepare() method runs last, similar to Django's form.clean().
+        This allows us to override anything we want. Here we add vocabularies
+        to the index because they must be dynamic.
+
+        Technically, we could handle the other stuff (run, course, etc.) here
+        as well, but don't because explicit is better than implicit.
+        """
+        prepared = super(LearningResourceIndex, self).prepare(obj)
+        for vocab in Vocabulary.objects.all():
+            # Values with spaces do not work, so these are replaced with
+            # underscores.
+            name = vocab.name.replace(" ", "_")
```

Comment:
Yes, because the backend treats it as multiple tokens. I did experiment with `slugify`, but it just complicated things slightly more and didn't add any benefit.